dpgen/data/gen.py

```diff
@@ -627,15 +653,41 @@ def run_vasp_md(jdata, mdata):
     run_tasks = [ii.replace(work_dir+"/", "") for ii in md_run_tasks]
     #dlog.info("md_work_dir", work_dir)
     #dlog.info("run_tasks",run_tasks)
-    dispatcher = make_dispatcher(mdata['fp_machine'], mdata['fp_resources'], work_dir, run_tasks, fp_group_size)
-    dispatcher.run_jobs(fp_resources,
-                       [fp_command],
-                       work_dir,
-                       run_tasks,
-                       fp_group_size,
-                       forward_common_files,
-                       forward_files,
-                       backward_files)
+    # dispatcher = make_dispatcher(mdata['fp_machine'], mdata['fp_resources'], work_dir, run_tasks, fp_group_size)
+    # dispatcher.run_jobs(fp_resources,
+    #                    [fp_command],
+    #                    work_dir,
+    #                    run_tasks,
+    #                    fp_group_size,
+    #                    forward_common_files,
+    #                    forward_files,
+    #                    backward_files)
+
+    fp_command = mdata['fp']['command']
+    task_list = []
+
+    for ii in run_tasks:
+        task = Task(command=fp_command, 
+            task_work_path=ii, 
+            forward_files=forward_files, 
+            backward_files=backward_files,
+        )
+        task_list.append(task)
+
+    machine = Machine.load_from_dict(mdata['fp']['machine'])
+    resources = Resources.load_from_dict(mdata['fp']['resources'])
+
+    submission = Submission(
+        work_base=work_dir,
+        machine=machine,
+        resources=resources,
+        forward_common_files=forward_common_files,
+        backward_common_files=[],
+        task_list=task_list
+    )
+
+    submission.run_submission()
```

Comment:
These codes are duplicated with others. I think it's better to create a new function which can be called by every task. It will be easier to maintain.